### DISCUSSION TITLE
Question about  Quasar and Vuejs new features

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#12627] [General - Components / Directives / etc]
Hi guys im working with vuejs since version 2.
By this time you had the options api , the vue cli and the vuex store.
Many things has been introduced since then , for instance:

composition api
script setup
vite 
pinia 
typescript support.
Other perhaps im missing.

I would like to have your opinion about the future of  quasar , vuejs and those features.
Are those mandatory / very important to learn ?
Other considerations , would be wellcome as well,
Regards.

**Comment by kamalsprasad** [comment]
As far as I can tell, composition api and script setup are the same, just the syntax is different. I really like script setup. Definitely recommend learning that. Of course, vanilla JS and Vue fundamentals are prerequisites. Vite looks cool but haven't played with it much. Pinia is the official recommendation for Vue store so it would be worth learning if you need it. Vuex store still works as does options api so you can stick with those if you like. Only reason I moved to script setup is because I started a new app and it made sense to go with that.

I think the future of quasar is bright like a quasar :). I LOVE IT!!! Updates come out on a regular basis. 

Hope this helps! ALl the best!
